runtime/src/main/java/org/corfudb/protocols/logprotocol/SMRRecord.java

```diff
@@ -0,0 +1,244 @@
+package org.corfudb.protocols.logprotocol;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+import lombok.Setter;
+import lombok.ToString;
+import org.corfudb.runtime.view.Address;
+import org.corfudb.util.serializer.ISerializer;
+import org.corfudb.util.serializer.Serializers;
+
+import java.util.Arrays;
+
+
+/**
+ * Created by mwei on 1/8/16.
+ */
+@SuppressWarnings("checkstyle:abbreviation")
+@ToString
+@EqualsAndHashCode(exclude = {"serializedSize"})
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+public class SMRRecord {
+
+    public final static SMRRecord COMPACTED_RECORD = SMRRecord.builder().build();
+    public final static byte[] COMPACTED_RECORD_SERIALIZED;
+
+    static {
+        ByteBuf buf = Unpooled.buffer(1);
+        COMPACTED_RECORD.serialize(buf);
+        COMPACTED_RECORD_SERIALIZED = buf.array();
+    }
+
+    /**
+     * The name of the SMR method. Note that this is limited to the size of a short.
+     */
+    @SuppressWarnings("checkstyle:MemberName")
+    @Getter
+    private String SMRMethod;
+
+    /**
+     * The arguments to the SMR method, which could be 0.
+     */
+    @SuppressWarnings("checkstyle:MemberName")
+    @Getter
+    private Object[] SMRArguments;
+
+    /**
+     * The serializer used to serialize the SMR arguments.
+     */
+    @Getter
+    private ISerializer serializerType;
+
+    /**
+     * An undo record, which can be used to undo this method.
+     */
+    @Getter
+    public transient Object undoRecord;
+
+    /**
+     * A flag indicating whether an undo record is present. Necessary
+     * because undo records may be NULL.
+     */
+    @Getter
+    public boolean undoable;
+
+    /**
+     * The upcall result, if present.
+     */
+    @Getter
+    public transient Object upcallResult;
+
+    /**
+     * If there is an upcall result for this modification.
+     */
+    @Getter
+    public transient boolean haveUpcallResult = false;
+
+    /**
+     * The locator of this instance of SMRRecord in global log.
+     */
+    @Getter
+    @Setter
+    public transient SMRRecordLocator locator = null;
+
+    /**
+     * The global address this record is associated to.
+     */
+    @Getter
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/app/annymartinez.m/CorfuDB/pullRequest?prid=4651450)